contracts/token-distribution/SaleRounds.sol

```diff
@@ -93,106 +115,39 @@ contract SaleRounds is TokenDistribution, GameOwner, ERC20 {
-        require(activeRound[roundType] == false, "Round is already active");
-        setActiveRoundInternally(roundType);
-    }
-
-    function setActiveRoundInternally(RoundType _roundType) private {
-        activeRound[_roundType] = true;
-        roundDistribution[_roundType].startTime = block.timestamp;
-    }
-
-    function addAddressForDistribution(string calldata _roundType, address _address) public
-        onlyGameOwner isRoundActive(_roundType) returns(bool) {
-
-        RoundType roundType = getRoundTypeByKey(_roundType);
-        addressMap[roundType][_address] = true;
         addressList[roundType].push(_address);
 
         return true;
     }
 
-    function deleteAddressForDistribution(string calldata _roundType, address _address, uint _index) public
-        onlyGameOwner isRoundActive(_roundType) returns(bool) {
+    function deleteAddressForDistribution(string calldata _roundType, address _address, uint _index) external
+        onlyGameOwner returns(bool) {
 
         RoundType roundType = getRoundTypeByKey(_roundType);
         require(_index < addressList[roundType].length, "index is out of distribution address array bounds");
         require(_address == addressList[roundType][_index], "Address does not match!");
 
-        addressMap[roundType][_address] = false;
         addressList[roundType][_index] = addressList[roundType][addressList[roundType].length - 1];
         addressList[roundType].pop();
         return true;
     }
 
-    function getAddressList(string calldata _roundType) public onlyGameOwner isRoundActive(_roundType) view returns(address[] memory){
+    function getAddressList(string calldata _roundType) external onlyGameOwner view returns(address[] memory){
         RoundType roundType = getRoundTypeByKey(_roundType);
         return addressList[roundType];
     }
 
     // @_amount is going be decimals() == default(18) digits
-    function reserveTokensInternal(RoundType _roundType, address _to, uint _amount) private {
-        require(roundDistribution[_roundType].supply >= _amount, "given amount is bigger than max supply for the round" );
-        require(roundDistribution[_roundType].totalRemaining >= _amount, "total remaining seed amount is not enough" );
-        roundDistribution[_roundType].totalRemaining -= _amount;
-        reservedBalances[_roundType][_to] += _amount;
-    }
-
-    // @_amount is going be decimals() == default(18) digits
-    function reserveTokens(string calldata _roundType, address _to, uint _amount) public
-        isRoundActive(_roundType) isInvestRound(_roundType) isEligibleToReserveToken(_roundType) {
+    function reserveTokens(string calldata _roundType, address _to, uint _amount) external
+    isInvestRound(_roundType) isEligibleToReserveToken(_roundType) {
         RoundType roundType = getRoundTypeByKey(_roundType);
 
         if(reservedBalances[RoundType.SEED][_to] > 0 && roundType != RoundType.SEED){
```

Comment:
I don't think both these checks are necessary. Since there is a separate mapping for each round, the same address can be in multiple rounds.

Reply:
Makes sense. I removed it.